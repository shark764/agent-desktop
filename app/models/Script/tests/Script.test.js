/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import Script from '../Script';

describe('Script', () => {
  describe('constructor', () => {
    it('should construct properly', () => {
      const script = new Script({
        id: 'special-id',
        name: 'special-name',
        elements: [{
          type: 'text',
          name: 'text-name',
        },{
          type: 'freeform',
          name: 'freeform-input-name',
          value: 'freeform-input-value',
        },{
          type: 'checkbox-input',
          name: 'checkbox-input-name',
          options: [{
            value: 'checkbox1',
          },{
            value: 'checkbox2',
          }],
        },{
          type: 'dropdown',
          name: 'dropdown-name',
        },{
          type: 'scale',
          name: 'scale-name',
        }],
        autoScriptDismiss: true,
        scriptReporting: true,
      });
      expect(script).toMatchSnapshot();
    });
  });
});
