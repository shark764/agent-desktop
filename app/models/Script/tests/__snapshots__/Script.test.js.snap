// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Script constructor should construct properly 1`] = `
Immutable.Record {
  "id": "special-id",
  "name": "special-name",
  "elements": Array [
    Object {
      "name": "text-name",
      "type": "text",
    },
    Object {
      "name": "freeform-input-name",
      "type": "freeform",
      "value": "freeform-input-value",
    },
    Object {
      "name": "checkbox-input-name",
      "options": Array [
        Object {
          "value": "checkbox1",
        },
        Object {
          "value": "checkbox2",
        },
      ],
      "type": "checkbox-input",
    },
    Object {
      "name": "dropdown-name",
      "type": "dropdown",
    },
    Object {
      "name": "scale-name",
      "type": "scale",
    },
  ],
  "values": Immutable.Map {
    "freeform-input-name": "freeform-input-value",
    "dropdown-name": null,
    "scale-name": null,
  },
  "autoScriptDismiss": true,
  "scriptReporting": true,
  "scrollTop": 0,
}
`;
